test/e2e/e2e.go

```diff
@@ -38,68 +38,110 @@ import (
 	_ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
 )
 
-const (
-	DefaultTestNamespace = "e2etest-knative-eventing"
+type clusterChannelProvisioner struct {
+	sync.Mutex
+	name string
+}
+
+func (ccp *clusterChannelProvisioner) Get() string {
+	ccp.Lock()
+	name := ccp.name
+	ccp.Unlock()
+	return name
+}
+
+func (ccp *clusterChannelProvisioner) Set(name string) {
```

Comment:
You should validate the name here for safety.

Reply:
Done.